compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp

```diff
@@ -634,6 +634,9 @@ static void enforceLayoutToMultiReductionOp(
     ArrayRef<DistributionLayout *> operandLattices,
     ArrayRef<const DistributionLayout *> resultLattices,
     std::function<void(DistributionLayout *, ChangeResult)> update) {
+  if (resultLattices.empty()) {
+    return;
+  }
```

Comment:
This shoudnt be part of this patch?

Reply:
It will crash for scalar case if this statement is not included, patch 2 mainly include the propagation from scf.for to scf.yield.

Reply:
ah ok, i see.